server/src/main/java/org/apache/lucene/queries/SearchAfterSortedDocQuery.java

```diff
@@ -66,7 +66,7 @@ public Weight createWeight(IndexSearcher searcher, ScoreMode scoreMode, float bo
             public Scorer scorer(LeafReaderContext context) throws IOException {
                 Sort segmentSort = context.reader().getMetaData().getSort();
                 if (segmentSort == null || Lucene.canEarlyTerminate(sort, segmentSort) == false) {
-                    throw new IOException("search sort :[" + sort.getSort() + "] does not match the index sort:[" + segmentSort + "]");
+                    throw new IOException("search sort :" + Arrays.toString(sort.getSort()) + " does not match the index sort:[" + segmentSort + "]");
```

Comment:
I think it would make sense use the Sort objects `toString()` method directly, the same way we print `segmentSort` later in the error message. It should print all the SortFields.

Reply:
But this is an array, and calling toString directly will only print out the address, which is not helpful to the user. You can use this code to verify:
        ```
SortField sortField = new SortField("name", SortField.Type.STRING);
        SortField[] sortFields = {sortField};
        System.out.println("sortFields:" + sortFields);  // sortFields:[Lorg.apache.lucene.search.SortField;@4cdbe50f
        System.out.println("sortFields:"+ Arrays.toString(sortFields));  // sortFields:[<string: "name">]
```
    Thanks

Reply:
I wasn't suggesting to print the array but the `sort` field, which is the same class as `segmentSort` and has a toString implementation. Its those two object we compare in Lucene.canEarlyTerminate anyway.

Reply:
I get it